waspc/.hlint.yaml

```diff
@@ -17,46 +6,10 @@
   - -XQuasiQuotes
   - -XScopedTypeVariables
 
-# Control which extensions/flags/modules/functions can be used
-#
-# - extensions:
-#   - default: false # all extension are banned by default
-#   - name: [PatternGuards, ViewPatterns] # only these listed extensions can be used
-#   - {name: CPP, within: CrossPlatform} # CPP can only be used in a given module
-#
-# - flags:
-#   - {name: -w, within: []} # -w is allowed nowhere
-#
-# - modules:
-#   - {name: [Data.Set, Data.HashSet], as: Set} # if you import Data.Set qualified, it must be as 'Set'
-#   - {name: Control.Arrow, within: []} # Certain modules are banned entirely
-#
-# - functions:
-#   - {name: unsafePerformIO, within: []} # unsafePerformIO can only appear in no modules
-
-
-# Add custom hints for this project
-#
-# Will suggest replacing "wibbleMany [myvar]" with "wibbleOne myvar"
-# - error: {lhs: "wibbleMany [x]", rhs: wibbleOne x}
-
-
-# Turn on hints that are off by default
-#
-# Ban "module X(module X) where", to require a real export list
-# - warn: {name: Use explicit module export list}
-#
-# Replace a $ b $ c with a . b $ c
-# - group: {name: dollar, enabled: true}
-#
-# Generalise map to fmap, ++ to <>
-# - group: {name: generalise, enabled: true}
-
-
-# Ignore some builtin hints
-# - ignore: {name: Use let}
-# - ignore: {name: Use const, within: SpecialModule} # Only within certain modules
-
-
-# Define some custom infix operators
-# - fixity: infixr 3 ~^#^~
+# TODO: Review the remaining ignored hlint rules.
+- ignore: {name: Use camelCase}
+- ignore: {name: Eta reduce}
+- ignore: {name: Use first} 
+- ignore: {name: Use second} 
+- ignore: {name: Use $>} 
+- ignore: {name: Use list comprehension} 
```

Comment:
Any specific reason you set these to ignored? Do you think they are not good, or didn't yet have time to go through these?

Reply:
I didn't want to make the judgement call on whether these changes improved the codebase.

Reply:
Fair enough! I will take a look at these then. Do you have any thoughts though?

Reply:
I'd like to not ignore any of them having seen the resulting linting errors. The list comprehension one is interesting. As for the eta reduction, if you think that it reduces readability then we could add a type signature above.

Reply:
Ok, so you are more or less for enabling all of them?
I do sometimes avoid eta-reduction on purpose in order to make the code more readable, that is true.
I don't see the benefits of eta-reduction though, it feels more cool than useful to me -> sometimes I do find it increases readibility, sometimes I find it decreases it. So I would be ok with not having this rule at all on one hand. On the other hand, if we did as you suggested, that is also fine at the end. I don't have a very strong opinion here.

Do you want to tackle all of these alone or do you want me to take something over? Let me know what/if you would like me to do!

Reply:
I'd like for the PR to me merged as-is and for the remaining TODOs to be tackled on a separate PR. When both are done we can close the related issue.